code/datums/actions/items/beserk.dm

```diff
@@ -0,0 +1,19 @@
+/datum/action/item_action/berserk_mode
+	name = "Berserk"
+	desc = "Increase your movement and melee speed while also increasing your melee armor for a short amount of time."
+	icon_icon = 'icons/mob/actions/actions_items.dmi'
+	button_icon_state = "berserk_mode"
+	background_icon_state = "bg_demon"
+
+/datum/action/item_action/berserk_mode/Trigger(trigger_flags)
+	if(istype(target, /obj/item/clothing/head/hooded/berserker))
```

Comment:
since you're returning at the end of this proc regardless, couldn't you just do uhh
```
if (!istype())
    return ..()
....
```